File: backend/app/utils/id_generator.py
import hashlib


def generate_news_id(url: str, title: str) -> str:
    """
    Generate unique ID from url and title using sha256.
    Returns first 24 characters of hash.
    """
    content = f"{url}|{title}"
    hash_object = hashlib.sha256(content.encode('utf-8'))
    return hash_object.hexdigest()[:24]

File: backend/src/utils/idGenerator.js
const crypto = require('crypto');

/**
 * Generate unique ID from url and title using sha256.
 * Returns first 24 characters of hash.
 */
function generateNewsId(url, title) {
  const content = `${url}|${title}`;
  const hash = crypto.createHash('sha256').update(content, 'utf8').digest('hex');
  return hash.slice(0, 24);
}

module.exports = { generateNewsId };
